docs/user/openstack/customization.md

```diff
@@ -137,6 +138,11 @@ platform:
       clusterOSImage: my-rhcos
 ```
 
+## Custom Subnets
+
+If you want to install your cluster on a custom network and subnet, you can pass the ID of the subnet to the install config with the `machinesSubnet` option. This subnet will be tagged as the primary subnet for the cluster. The nodes, and VIP port will be created on this subnet. This does add some complexity to the install process that needs to be accounted for. First of all, the CIDR of the custom subnet must match the cidr of `networks.machineNetwork`. Also be aware that the API and Ingress VIPs by default will try to take the .5 and .7 of your network CIDR. If they are not available, then you should set them in the install config by setting the
```

Comment:
The user-provided subnet needs to have a DHCP server. This will likely always be the case, but better be explicit about the requirement.

Also, we should explain how to avoid potential IP address conflicts with the API and Ingress VIPs. These IP should be picked outside of the DHCP range and passed to the installer via the `apiVIP` and `ingressVIP` options. It would actually be better if we could let the DHCP assign the VIPs on the ports.

Reply:
How would we have dhcp assign the ports? I looked into allowing the ports to be set dynamically in the installer and the way things are right now, it is not doable.

Reply:
Using the CLI, by specifying `--fixed-ip "subnet=<my-subnet>"`, it's possible to let the DHCP assign an IP to the port:
```
$ openstack port create --network mandre-krc6g-openshift --security-group mandre-krc6g-worker --fixed-ip "subnet=mandre-krc6g-nodes" --tag openshiftClusterID="mandre-krc6g" "mandre-krc6g-test-port"
$ openstack port show mandre-krc6g-test-port -f value -c fixed_ips                                                                                                                                       
[{'subnet_id': 'c72ac7d7-2f46-4698-92ad-22458089de7d', 'ip_address': '10.0.128.34'}]
```

Reply:
the installer doesn't let you pass a port, it only lets you pass it an IP address to make a port from. I am proposing changing it to allow you to pass the ports directly here though: https://github.com/openshift/enhancements/pull/342

Reply:
Well, the installer creates the port. It's the same thing. I showed how it worked with the CLI. In terraform, you'd have to pass a `fixed_ip` stuct with a `subnet_id` for the port creation.

https://www.terraform.io/docs/providers/openstack/r/networking_port_v2.html#fixed_ip